private/bufpkg/bufimage/bufimagemodify/internal/file_options_trie_test.go

```diff
@@ -0,0 +1,342 @@
+			)
+		})
+	}
+}
+
+func TestRegisterDescendant(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		description                      string
+		pathsToInsert                    [][]int32
+		pathsToRegister                  [][]int32
+		expectedIndicesWithoutDescendant []int
+	}{
+		{
+			description: "register none",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},       // 0
+				{4, 0, 2, 1, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				1,
+				2,
+			},
+		},
+		{
+			description: "register non-child",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},       // 0
+				{4, 0, 2, 1, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			pathsToRegister: [][]int32{
+				{30},
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				1,
+				2,
+			},
+		},
+		{
+			description: "register sibling",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},       // 0
+				{4, 0, 2, 1, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			pathsToRegister: [][]int32{
+				{40, 0, 2, 0, 1},
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				1,
+				2,
+			},
+		},
+		{
+			description: "register child",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 3, 8},       // 0
+				{4, 0, 2, 2, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			pathsToRegister: [][]int32{
+				{4, 0, 2, 2, 8, 6}, // descendant of 1
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				2,
+			},
+		},
+		{
+			description: "register descendant",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},       // 0
+				{4, 0, 2, 2, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			pathsToRegister: [][]int32{
+				{4, 0, 2, 2, 8, 0, 1139}, // descendant of 1
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				2,
+			},
+		},
+		{
+			description: "register multiple for the same parent",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},       // 0
+				{4, 0, 2, 1, 8},       // 1
+				{4, 0, 3, 0, 2, 1, 8}, // 2
+			},
+			pathsToRegister: [][]int32{
+				{4, 0, 2, 1, 8, 0, 1139}, // descendant of 1
+				{4, 0, 2, 1, 8, 6},       // descendant of 1
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				2,
+			},
+		},
+		{
+			description: "register for multiple parents",
+			pathsToInsert: [][]int32{
+				{4, 0, 2, 0, 8},             // 0, no register
+				{4, 0, 2, 1, 8},             // 1
+				{4, 0, 2, 2, 8},             // 2
+				{4, 0, 3, 0, 2, 1, 8},       // 3, no register
+				{4, 0, 3, 0, 3, 1, 2, 3, 8}, // 4
+				{7, 0, 8},                   // 5
+				{7, 1, 8},                   // 6, no register
+			},
+			pathsToRegister: [][]int32{
+				{4, 0, 3, 0, 3, 1, 2, 3, 8, 1},
+				{4, 0, 2, 1, 8, 0, 1139},
+				{4, 0, 2, 1, 8, 6},
+				{7, 0, 8, 50003, 0},
+				{4, 0, 2, 2, 8, 5},
+			},
+			expectedIndicesWithoutDescendant: []int{
+				0,
+				3,
+				6,
+			},
+		},
+		{
+			description: "register more none-field option",
```

Comment:
Might be good to have a case where expected indices are empty (i.e. all option paths retain at least one descendant)?